docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -0,0 +1,1362 @@
+   </td>
+  </tr>
+  <tr>
+   <td><code>TrainingRuntimeRef</code>
+   </td>
+   <td>Reference to the existing <code>TrainingRuntime</code> that pre-deployed by platform engineers
+   </td>
+  </tr>
+  <tr>
+   <td><code>TrainerConfig</code>
+   </td>
+   <td>Configuration for the Trainer such as image, number of nodes, accelerators.
+   </td>
+  </tr>
+  <tr>
+   <td><code>ModelConfig</code>
+   </td>
+   <td>Configuration for the pre-trained model and location for model output
+   </td>
+  </tr>
+  <tr>
+   <td><code>DatasetConfig</code>
+   </td>
+   <td>Configuration for the dataset that will be used to train or fine-tune model
+   </td>
+  </tr>
+  <tr>
+   <td>Labels and Annotations
+   </td>
+   <td>Custom metadata that needs to be applied to the <code>TrainJob</code> resources: JobSet, Job, Pods.
+   </td>
+  </tr>
+  <tr>
+   <td><code>PodSpecOverrides</code>
+   </td>
+   <td>Custom overrides that are specific to the <code>TrainJob</code> and need to be applied to the
+    <code>TrainJob</code> resources. For example, the user identity. Usually, it is managed by
+    custom admission webhooks that inject data to the <code>TrainJob</code> after user creates it
+    via Python SDK or <code>kubectl</code>
+   </td>
+  </tr>
+</table>
+
+### Example of TrainJob
+
+```yaml
+apiVersion: kubeflow.org/v2alpha1
+kind: TrainJob
+metadata:
+  name: torch-ddp
+  namespace: tenant-alpha
+spec:
+  trainingRuntimeRef:
+    name: torch-distributed-multi-node
+  trainerConfig:
+    image: docker.io/custom-training
+    command:
+      - torchrun train.py
+    numNodes: 5
+    resourcesPerNode:
+      requests:
+        nvidia.com/gpu: 2
+```
+
+The above command will be converted as follows:
+
+```bash
+torchrun --nnodes=5 --nproc-per-node=2 train.py
+```
+
+Additionally, the Kubeflow Training SDK allows to create the above `TrainJob` using the Python API:
+
+```python
+def train_func():
+    import torch
+    class Net(torch.nn.Module):
+        """Create the PyTorch Model"""
+        ...
+    model = Net()
+
+    # Attach model to the distributor
+    torch.distributed.init_process_group(backend="nccl")
+    model = torch.nn.parallel.DistributedDataParallel(model)
+
+    # Train model
+    model.train()
+
+# Use Kubeflow SDK to create TrainJob.
+from kubeflow.training import TrainingClient
+
+TrainingClient().train(
+    name="torch-ddp",
+    func=train_func,
+    num_nodes=5,
+    resources_per_node={"gpu": 2},
+)
+```
+
+### Example of LLM Fine-Tuning
+
+This example shows how to create `TrainJob` to fine-tune LLama 7b:
+
+```yaml
+apiVersion: kubeflow.org/v2alpha1
+kind: TrainJob
+metadata:
+  name: tune-llama-with-yelp
+  namespace: tenant-alpha
+spec:
+  trainingRuntimeRef:
+    name: torch-tune-llama-7b
+  datasetConfig:
+    s3:
+      bucket: datasets
+      path: custom-datasets/yelp-review
+```
+
+### The Trainer Config API
+
+The `TrainerConfig` represents the APIs that data scientists can use to configure trainer settings:
+
```

Comment:
Yes, but we also implement Python API to manage TrainJobs. It can be used as in this example: https://www.kubeflow.org/docs/components/training/getting-started/#getting-started-with-pytorchjob.

Additionally, we want to add UI/Jupyter Plugin to monitor Jobs.